libkbfs/mdcache.go

```diff
@@ -111,3 +117,30 @@ func (md *MDCacheStandard) MarkPutToServer(
 	rmd.putToServer = true
 	md.lru.Add(key, rmd)
 }
+
+// GetIDForHandle implements the MDCache interface for
+// MDCacheStandard.
+func (md *MDCacheStandard) GetIDForHandle(handle *TlfHandle) (tlf.ID, error) {
+	md.lock.RLock()
+	defer md.lock.RUnlock()
+	key := handle.GetCanonicalPath()
+	tmp, ok := md.idLRU.Get(key)
+	if !ok {
+		return tlf.NullID, errors.WithStack(NoSuchTlfIDError{handle})
+	}
+	id, ok := tmp.(tlf.ID)
+	if !ok {
+		return tlf.NullID, errors.Errorf("Bad ID for handle %s", key)
```

Comment:
Shouldn't we wipe it from the cache and return a `NoSuchTlfIDError`?

Reply:
I was being consistent with the other Get method in this class.  Also, if there's a type error like this I'd rather know about it than hide it, when we're talking about the client side.  I'd be happy to panic here if you prefer.